packages/esm-patient-chart-app/src/patient-chart/patient-chart.resources.ts

```diff
@@ -0,0 +1,20 @@
+import { openmrsFetch, restBaseUrl, type Visit } from '@openmrs/esm-framework';
+import useSWR from 'swr';
+
+const defaultVisitCustomRepresentation =
+  'custom:(uuid,display,voided,indication,startDatetime,stopDatetime,' +
+  'encounters:(uuid,display,encounterDatetime,' +
+  'form:(uuid,name),location:ref,' +
+  'encounterType:ref,' +
+  'encounterProviders:(uuid,display,' +
+  'provider:(uuid,display))),' +
+  'patient:(uuid,display),' +
+  'visitType:(uuid,name,display),' +
+  'attributes:(uuid,display,attributeType:(name,datatypeClassname,uuid),value),' +
+  'location:(uuid,name,display))';
+
+export function useVisitByUuId(visitUuid: string, representation: string = defaultVisitCustomRepresentation) {
+  const url = `${restBaseUrl}/visit/${visitUuid}?v=${representation}`;
+
+  return useSWR<{ data: Visit }>(visitUuid ? url : null, openmrsFetch);
```

Comment:
```suggestion
  const { data, error, isLoading, isValidating, mutate } = useSWR<{ data: Visit }>(key, openmrsFetch);

  return {
    visit: data?.data,
    error,
    isLoading,
    isValidating,
    mutate,
  };
```

Nitpicky, but it's more canonical to expose destructured, domain-specific fields instead of handing that responsibility to callers.